### PR TITLE
use compiletest_rs

### PR DESCRIPTION
**Pull request by oli-obk**
This allows error checking with rustc's `//~ERROR: ` comments

### CODE REVIEW

**Comment by tsion** [comment]
Nice, I really wanted to have that!

File: src/bin/miri.rs [hunk 1]
```diff
@@ -6,14 +6,13 @@ extern crate rustc_driver;
...
 use rustc::session::Session;
-use rustc_driver::{driver, CompilerCalls, Compilation};
+use rustc_driver::{driver, CompilerCalls};
 
 struct MiriCompilerCalls;
 
 impl<'a> CompilerCalls<'a> for MiriCompilerCalls {
     fn build_controller(&mut self, _: &Session) -> driver::CompileController<'a> {
         let mut control = driver::CompileController::basic();
-        control.after_analysis.stop = Compilation::Stop;
```
Comment (tsion) [inline src/bin/miri.rs]: Why did you remove this? Does this mean trans (and any other post-analysis steps) will run after Miri finishes?

Comment (oli-obk) [inline src/bin/miri.rs]: yea. We need that to make run-pass tests actually runnable

Comment (oli-obk) [inline src/bin/miri.rs]: there are no "compile-suceed" tests: https://github.com/laumann/compiletest-rs/blob/master/src/common.rs#L18-L28

Comment (tsion) [inline src/bin/miri.rs]: Ah, good point. I guess technically it doesn't make sense for any Miri tests to be run-pass because the execution all happens at compile-time, but this is fine for now.

Later we can rewrite those tests in a smarter way.

Comment (oli-obk) [inline src/bin/miri.rs]: rust has lots of run-pass tests that don't do anything but verify that some construct will compile. So there's precedent for that ;)

Comment (tsion) [inline src/bin/miri.rs]: Fair enough. Miri needs *much* better tests, regardless. :P
